src/core/common/instance.hpp

```diff
@@ -640,6 +640,13 @@ template <> inline NetworkData::Notifier &Instance::Get(void)
 }
 #endif
 
+#if OPENTHREAD_CONFIG_NETDATA_PUBLISHER_ENABLE
+template <> inline NetworkData::Publisher &Instance::Get(void)
+{
+    return mThreadNetif.mNetworkDataPublisher;
```

Comment:
There seems to be only one `Publisher`? How can we publish multiple items (both SRP unicast and external routes)?

Reply:
The `Publisher` class can provide different methods/APIs for publishing different things. Same way that we have one `NetworkData::Local` object but can use it to add service and/or on-mesh/external-route prefix.

Reply:
But it seems to only be able to publish one thing at a time? 
Should we move Publisher instance to be a member of Srp::Server?

Reply:
> But it seems to only be able to publish one thing at a time?

In current code, yes! We can extend it later. Check this branch [abtink/netdata/publisher-full](https://github.com/abtink/openthread/tree/netdata/publisher-full) if you are interested to see how (this is from earlier with on-mesh/off mesh support treated as individual entries)

----

> Should we move Publisher instance to be a member of Srp::Server?

`Publisher` can be used independent of whether OT SRP server feature is being used or not (e.g. in cases where app/next layer is providing SRP sever functionality itself).